overlord/devicestate/devicestate.go

```diff
@@ -488,26 +513,43 @@ func remodelEssentialSnapTasks(ctx context.Context, st *state.State, pathSI *pat
 		return nil, err
 	}
 
+	revOpts := &snapstate.RevisionOptions{
+		Channel:  newModelSnapChannel,
+		Revision: ms.newSnapRevision,
+	}
+
+	if !revOpts.Revision.Unset() {
+		revOpts.ValidationSets = ms.newModelValidationSets
+	}
+
```

Comment:
this code is not particularly complex but we repeat it in a couple of places, I simple helper function might still be a good idea